libres/lib/python/init.cpp

```diff
@@ -0,0 +1,22 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <vector>
+#include <iostream>
+#include <ert/enkf/obs_vector.hpp>
+
+namespace py = pybind11;
+
+obs_vector_type* to_obs_vector_type(py::object obj) {
+    py::object address = obj.attr("_address")();
+    void *pointer = PyLong_AsVoidPtr(address.ptr());
+    return obs_vector_safe_cast(pointer);
+}
+
+const std::vector<int>& py_obs_vector_get_step_list(py::object self) {
+    obs_vector_type* obs_vector = to_obs_vector_type(self);
+    return obs_vector_get_step_list(obs_vector);
+}
+
+PYBIND11_MODULE(_lib, m) {
+    m.def("obs_vector_get_step_list", &py_obs_vector_get_step_list);
```

Comment:
Style suggestion: Use a lambda expression rather than a full-blown function.
```suggestion
PYBIND11_MODULE(_lib, m) {
    m.def("obs_vector_get_step_list", [](py::object self) {
        auto obs_vector = to_obs_vector_type(self);
        return obs_vector_get_step_list(obs_vector);
    });
```

Reply:
Additionally, we might want to tell Python that the first argument's name is `self` rather than the pybind11 default `arg0`. This appears in `ArgumentError`s thrown by pybind11, as well as letting us use kwargs (not relevant in this case but still).
```suggestion
PYBIND11_MODULE(_lib, m) {
    m.def("obs_vector_get_step_list", [](py::object self) {
        auto obs_vector = to_obs_vector_type(self);
        return obs_vector_get_step_list(obs_vector);
    }, py::arg("self"));
```